File: Java/1897. Redistribute Characters to Make All Strings Equal.java
class Solution {
    public boolean makeEqual(String[] words) {
        Map<Character, Integer> chars = new HashMap<>();
        for (String w: words) {
            for (char c: w.toCharArray()) {
                chars.put(c, chars.getOrDefault(c, 0) + 1);
            }
        }

        int total = words.length;
        for (int count: chars.values()) {
            if (count % total > 0) {
                return false;
            }
        }

        return true;
    }
}

File: Python3/1897. Redistribute Characters to Make All Strings Equal.py
class Solution:
    def makeEqual(self, words: List[str]) -> bool:
        chars = Counter()
        for w in words:
            for c in w:
                chars[c] += 1

        total = len(words)
        for count in chars.values():
            if count % total:
                return False

        return True